.github/workflows/model-evaluation.yml

```diff
@@ -0,0 +1,201 @@
+name: Full Model Evaluation
+
+on:
+  workflow_dispatch:
+    inputs:
+      provision-and-deploy:
+        description: 'Whether to provision the deployment'
+        required: true
+        default: true
+        type: boolean
+      deploy-only:
+        description: 'Whether to deploy the application'
+        required: true
+        default: false
+        type: boolean
+      purge:
+        description: 'Whether to purge the deployment after evaluation'
+        required: true
+        default: false
+        type: boolean
+
+permissions:
+  id-token: write
+  contents: read
+
+jobs:
+    deploy-azure:
+      if: ${{ inputs.provision-and-deploy == true || inputs.deploy-only == true }}
+      uses: ./.github/workflows/azure-dev.yml
+      secrets: inherit
+      with:
+        deploy-only: ${{ inputs.deploy-only }}
```

Comment:
I will look for a method to automate this, I might implement this inside the evaluation scripts or in a separate python script file, for each different variables there might be different way to retrieve their value. 

However, please allow me to clarify here, using our current configuration and the way we do the deployment each time, I can confirm the values are always identical regardless how many times it's redeployed (or provisioning). This is not "configured at some arbitrary point in time", you can think this as an initial configuration that needs to be done.

Reply:
> I will look for a method to automate this, I might implement this inside the evaluation scripts or in a separate python script file, for each different variables there might be different way to retrieve their value.

No, there already is a way. Isn't a `.env` file created with all of the environment variables? That should be all we need. You need to ensure that this file is carried over to the next job or that its content is sourced into it.

> However, please allow me to clarify here, using our current configuration and the way we do the deployment each time, I can confirm the values are always identical regardless how many times it's redeployed (or provisioning).

Yes, they are, because we generate all names based on the name of the resource group (or less often the subscription). But what if something changes in the code? Why would someone have to update repo configurations following a change? They wouldn't as things like these create inconsistencies. We set environment variables and secrets on GH for things that are actually **constant**. And these are just the few values in our environment that have to be there when we first deploy: the environment name, the subscription ID, the location for all resources, the SKU and location for Search, and the location of Document Intelligence. There's no point in hardcoding everything else, they'll be populated once the provisioning succeeds or once the tool detects that the desired resources are already there.

> This is not "configured at some arbitrary point in time", you can think this as an initial configuration that needs to be done.

This is wrong because you don't know these names before deploying a specific version of the code. You don't know how these hashes included in the names are generated. In fact, without looking at the code you can't even know for sure that they're hashes. What if we change the way we name resources in the code? Even better, what if Bicep changes their hashing algorithm? You can't assume that all of these values will stay the same. If you wanted this to make sense, you could have selected manually picked names for all of these resources. But you really don't have to, because we really do want to just run the provisioning script every time and propagate all of the envvars it created to the next jobs/steps.

Reply:
okay, I will use variables in the `.env` file to set the default values for eval script configs.